WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+JetpackInstall.swift

```diff
@@ -2,34 +2,66 @@ import UIKit
 
 extension MySiteViewController {
     func subscribeToJetpackInstallNotifications() {
-        NotificationCenter.default.addObserver(self, selector: #selector(launchJetpackInstallFromNotification(_:)), name: .installJetpack, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(promptJetpackPluginInstallFromNotification(_:)), name: .promptInstallJetpack, object: nil)
     }
 
-    /// Presents the Jetpack Login (Install) controller from a notification
-    @objc func launchJetpackInstallFromNotification(_ notification: NSNotification) {
-        guard let blog = blog else {
+    /// Presents the prompt to install Jetpack from a notification
+    @objc func promptJetpackPluginInstallFromNotification(_ notification: NSNotification) {
+        guard let blog = blog,
+            let promptSettings = notification.userInfo?["jetpackInstallPromptSettings"] as? JetpackInstallPromptSettings else {
             return
         }
 
-        let controller = JetpackLoginViewController(blog: blog)
-        controller.promptType = .installPrompt
+        let installPromptViewController = JetpackInstallPromptViewController(blog: blog)
+        let installPromptNavigationController = UINavigationController(rootViewController: installPromptViewController)
+        installPromptNavigationController.modalPresentationStyle = .fullScreen
 
-        let navController = UINavigationController(rootViewController: controller)
-        navController.modalPresentationStyle = .fullScreen
+        installPromptViewController.actionHandler = { [weak self] action in
+            switch action {
+            case .noThanks:
+                installPromptNavigationController.dismiss(animated: true)
+            case .install:
+                self?.presentJetpackInstall(on: installPromptNavigationController, blog: blog, jetpackInstallPromptSettings: promptSettings)
+            }
+        }
 
-        navigationController?.present(navController, animated: true)
+        navigationController?.present(installPromptNavigationController, animated: true)
```

Comment:
```suggestion
        present(installPromptNavigationController, animated: true)
```

Is there an advantage to presenting from the nav controller rather than its root view controller?

Reply:
> I notice that the "what would you like to focus on" question is shown after Jetpack setup is complete. This might be out of scope, but I didn't see it on the link to the known "existing issues".

Yes, it appears after WP.com login. As I understood it's mentioned under `The login epilogue can be displayed during the connection process, we should prevent it from displaying`

> Is there an advantage to presenting from the nav controller rather than its root view controller?

There's no advantage or disadvantage. Those 2 calls are equivalent, since MySiteViewController is inside `UINavigationController`, so other controllers are presented in & on `UINavigationController` either way. I'll remove the redundant `navigationController?.` code.

> When I finish the whole setup, both the installation flow and the prompt flow modals are dismissed in quick succession. Should we instead dismiss both modals in one single animation?
Are there too many modals? In case 2, the user dismisses two modals and the app dismisses another (total 3 modals).

Yes, it would look better 2 dismiss them with a single animation.

It's a good question if there're too many modals. Installation flow by itself is designed to be presented in a modal (with all the dismissal cancellation buttons in the navigation bar) and WP.com login flow is presented as a modal as well. 

I'll check what changes I'd need to make if we push the installation flow instead of presenting it on top of the installation prompt.

Reply:
> Yes, it appears after WP.com login. As I understood it's mentioned under `The login epilogue can be displayed during the connection process, we should prevent it from displaying`

I'd interpreted the epilogue as being the site picker, but I agree it also covers this focus question screen.

> There's no advantage or disadvantage. Those 2 calls are equivalent, since MySiteViewController is inside `UINavigationController`, so other controllers are presented in & on `UINavigationController` either way. I'll remove the redundant `navigationController?.` code.

Thanks for explaining, I wasn't familiar with calling `present` from a navigation controller.

> I'll check what changes I'd need to make if we push the installation flow instead of presenting it on top of the installation prompt.

Ok 👍